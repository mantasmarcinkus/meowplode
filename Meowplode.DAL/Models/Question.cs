using System.Collections.Generic;

namespace Meowplode.DAL.Models
{
    public class Question : BaseModel
    {
        public string Value { get; set; }

        public List<Option> Options { get; set; }
    }
}
