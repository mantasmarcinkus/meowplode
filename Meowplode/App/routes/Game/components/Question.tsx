import * as React from "react";
import { ActionCreators, DefinedState } from "../module";
const KeyHandler = (require("react-key-handler") as any).default;
const KEYPRESS = (require("react-key-handler") as any).KEYPRESS;

interface Props extends Pick<DefinedState & ActionCreators,
  "question" | "endGame" | "lastQuestion" | "providedResults" | "name" | "submitOption"> { }

export default class Question extends React.Component<Props, {}> {
  constructor() {
    super();
    this.handleKeyPress = this.handleKeyPress.bind(this);
    this.onOptionSubmit = this.onOptionSubmit.bind(this);
  }
  public render() {
    const options = this.props.question.options.map(option => (
      <span key={option.id} style={{ display: "block" }}>
        <button
          className="answer"
          value={option.id}
          onClick={this.onOptionSubmit}
        >
          {option.value}
        </button>
      </span>),
    );
    const keyHandlers = this.props.question.options.map((option, index) => (
      <KeyHandler
        key={index}
        keyEventName={KEYPRESS}
        keyValue={(index + 1).toString()}
        onKeyHandle={this.handleKeyPress}
      />),
    );
    return (
      <div>
        {keyHandlers}
        <div className="question">{this.props.question.value}</div>
        {options}
      </div>
    );
  }
  private handleKeyPress({ key }: { key: number }) {
    this.onOptionSubmit({ target: { value: this.props.question.options[key - 1].id } });
  }
  private onOptionSubmit({ target: { value } }: any) {
    if (this.props.lastQuestion) {
      this.props.endGame({
        name: this.props.name,
        providedResults: [
          ...this.props.providedResults,
          { questionId: this.props.question.id, optionId: value },
        ],
      });
    }

    this.props.submitOption(this.props.question.id, value);
  }
}
