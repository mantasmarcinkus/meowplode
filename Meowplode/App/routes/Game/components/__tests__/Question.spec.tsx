import { shallow, ShallowWrapper } from "enzyme";
import * as React from "react";
import Question from "../Question";

jest.mock("../../../Leaderboard/container");

// as any. good for injection, bad for testing, should be filled though
const testProps: any = {
  imageSource: "image",
  lastQuestion: false,
  name: "text",
  providedResults: [],
  question: {
    id: "1",
    options: [
      { id: "2", value: "opt1" },
      { id: "3", value: "opt2" },
    ],
    value: "val",
  },
};

describe("<Question />", () => {
  let wrapper: ShallowWrapper<any, any>;
  beforeEach(() => {
    wrapper = shallow(<Question {...testProps} />);
  });
  it("renders", () => {
    expect(wrapper.render()).toMatchSnapshot();
  });
});
