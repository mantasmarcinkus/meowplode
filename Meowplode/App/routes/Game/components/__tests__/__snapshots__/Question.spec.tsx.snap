// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Question /> renders 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "attribs": Object {
              "class": "question",
            },
            "children": Array [
              Object {
                "data": "val",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "next": Object {
              "attribs": Object {
                "style": "display:block;",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "answer",
                    "value": "2",
                  },
                  "children": Array [
                    Object {
                      "data": "opt1",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                },
              ],
              "name": "span",
              "next": Object {
                "attribs": Object {
                  "style": "display:block;",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "answer",
                      "value": "3",
                    },
                    "children": Array [
                      Object {
                        "data": "opt2",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "button",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                  },
                ],
                "name": "span",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
          },
          Object {
            "attribs": Object {
              "style": "display:block;",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "answer",
                  "value": "2",
                },
                "children": Array [
                  Object {
                    "data": "opt1",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
              },
            ],
            "name": "span",
            "next": Object {
              "attribs": Object {
                "style": "display:block;",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "answer",
                    "value": "3",
                  },
                  "children": Array [
                    Object {
                      "data": "opt2",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                },
              ],
              "name": "span",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "question",
              },
              "children": Array [
                Object {
                  "data": "val",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
            },
            "type": "tag",
          },
          Object {
            "attribs": Object {
              "style": "display:block;",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "answer",
                  "value": "3",
                },
                "children": Array [
                  Object {
                    "data": "opt2",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "button",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
              },
            ],
            "name": "span",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "style": "display:block;",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "answer",
                    "value": "2",
                  },
                  "children": Array [
                    Object {
                      "data": "opt1",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "button",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                },
              ],
              "name": "span",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "question",
                },
                "children": Array [
                  Object {
                    "data": "val",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
              },
              "type": "tag",
            },
            "type": "tag",
          },
        ],
        "name": "div",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;
